qual/src/main/java/com/virtuslab/qual/guieffect/UIThreadUnsafe.java

```diff
@@ -5,9 +5,26 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-// TODO (typetools/checker-framework#3252): replace with proper @Heavyweight
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
-// Must have runtime retention to be visible to ArchUnit tests.
+/**
+ * Used to mark methods that should NOT be executed on UI thread
+ * (in case of IntelliJ Platform, on Swing's Event Dispatch Thread aka EDT)
+ * to avoid a UI freeze.
```

Comment:
```suggestion
 * Used to mark methods that should NOT be executed on UI thread to avoid a UI freeze.
 * In case of IntelliJ Platform, the UI thread is Swing's Event Dispatch Thread aka EDT.
```